libs/util/data-fetcher/src/lib/sql-utils.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  FieldAggregation,
+  FieldFilter,
+  FieldGroupBy,
+  FieldName,
+  FieldSort,
+} from './model'
+
+function filterToSql(filter: FieldFilter): string {
+  const operator = filter[0]
+  const args = filter.slice(1)
+  function valueToSql(value: string | number) {
+    return typeof value === 'number' ? value : `'${value}'`
+  }
+  switch (operator) {
+    case '<':
+    case '<=':
+    case '>':
+    case '>=':
+    case '=':
+    case '!=':
+    case 'like':
+      return `[${args[0]}] ${operator.toUpperCase()} ${valueToSql(
+        args[1] as string | number
+      )}`
+    case 'in': {
+      const values = args.slice(1) as string[] | number[]
+      return `[${args[0]}] IN (${values.map(valueToSql).join(', ')})`
+    }
+    case 'and':
+    case 'or': {
+      const children = (args as FieldFilter[])
+        .map(filterToSql)
+        .join(` ${operator.toUpperCase()} `)
+      return `(${children})`
+    }
+    case 'not':
+      return `NOT (${filterToSql(args[0] as FieldFilter)})`
+  }
+  throw new Error(
+    `Could not generate SQL query, operator not recognized: ${operator}`
+  )
+}
+
+function aggregationToSql(aggregation: FieldAggregation): string {
```

Comment:
ok ... note that aggregation should auto imply a groupby

Reply:
The way I see things is that we either:
* do a normal query without grouping, and then aggregations do not make much sense
* do an aggregation query, and then grouping by something (even if it's all records into one row) is required

The two approaches are mutually exclusive, even if in SQL there are actually more possibilities than that.

Reply:
sorry, I misread your comment, I guess we agree on that :)

Reply:
well conceptually this could be expressed like so:
```
dataset
  .groupBy(['dateTrunc', 'date', 'day'], ['distinct', 'nom'])
  .aggregate(['sum', 'valeur'])
  .orderBy([asc, 'date'])
```

What's going to be tricky is to have the 'dateTrunc' operation work across different sources.